### PR TITLE
Update README to tell specifically about Esprima-FB

### PR DESCRIPTION
**Pull request by RReverser**
- Added info on Esprima-FB with corresponding links to spec.
- Updated reference links to AST and tests pages.
- Added short explanation on versioning based on @zpao's comment: https://github.com/facebook/esprima/issues/35#issuecomment-52277629

Closes #40.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,20 +5,27 @@ parser written in ECMAScript (also popularly known as
 Esprima is created and maintained by [Ariya Hidayat](http://twitter.com/ariyahidayat),
 with the help of [many contributors](https://github.com/ariya/esprima/contributors).
 
+**Esprima-FB** is a fork of [Harmony branch](https://github.com/ariya/esprima/tree/harmony) of Esprima that implements [JSX specification](https://github.com/facebook/jsx) on top of ECMAScript syntax.
```
Comment (zpao) [inline README.md]: Nit "...the Harmony branch..."

Comment (RReverser) [inline README.md]: Though it makes sense to note that it's upstream is not Esprima's master but Harmony branch (which powers ES6 features).

Comment (zpao) [inline README.md]: I just want "the" inserted so the sentence reads better.

Comment (RReverser) [inline README.md]: Oh, sorry, will add :blush:

Comment (RReverser) [inline README.md]: @zpao Done.

**Pull request merged**